src/eigen.jl

```diff
@@ -6,12 +6,14 @@ end
 @inline function eigvals(a::StaticArray; permute::Bool=true, scale::Bool=true)
     if ishermitian(a)
         _eigvals(Size(a), Hermitian(a), permute, scale)
+    elseif Size(a) == Size{(1,1)}()
+        _eigvals(Size(a), a, permute, scale)
     else
         error("Only hermitian matrices are diagonalizable by *StaticArrays*. Non-Hermitian matrices should be converted to `Array` first.")
     end
 end
 
-@inline _eigvals(::Size{(1,1)}, a, permute, scale) = @inbounds return SVector(real(a.data[1]))
+@inline _eigvals(::Size{(1,1)}, a, permute, scale) = @inbounds return SVector(a.data[1])
```

Comment:
Rather than `a.data[1]`, I think this should be `Tuple(a)` which is now the interface for obtaining the tuple of elements, regardless of whether `a` is an `SMatrix` or some other type of static array.